### ISSUE TITLE
from html5slides to io2012

### ISSUE THREAD

**Issue by ptoche** [ramnathv/slidify#409]
I got a set of slides tweaked under the html5slides framework until I realized that io2012 is a more recent incarnation, so I thought I should go with the times and adopt i02012. Several remarks. Please let me know if I'm doing anything wrong or if my interpretations are incorrect.

  1. Under html5slides, if I dump any number of css files into assets/css they will be sourced.

  2. But io2012 does not source the assets/css directory.

  3. With io2012, your css must/can be stored in libraries/frameworks/io2012/css and you must/can edit the head.html template in libraries/frameworks/io2012/partials/head.html by inserting a reference to the css file, after the other ones already listed there:   

    <link rel="stylesheet" href="{{page.url.framework}}/css/custom.css"/>

where custom.css is the name of your css.

  4. With io2012, your css cannot be named io2012.css, but it can be named custom.css (not tested but perhaps numbers are not allowed in the name)

  5. If you want to insert a background image and refer to it in your css with 

    background-image: url(bg.png)

where bg.png is the name of your image and it must be stored inside libraries/frameworks/io2012/css
If you store the image in the more natural libraries/frameworks/io2012/images, it won't be found with the above reference.

6. If you want to remove the greyish color at the bottom of the slide (it looks very ugly when you have figures and figure captions with a white background that cover it only partially), you must/can add this to your css (the next-to-last line was the crucial one, but just to be safe I copied the whole code chunk from the default.css and painted it white all over):

    /* Remove color gradient at bottom of slides */
    slides > slide.backdrop {
      z-index: -10;
      display: block !important;
      background: -webkit-gradient(linear, 50% 0%, 50% 100%, color-stop(0%, #ffffff), color-stop(85%, #ffffff), color-stop(100%, #ffffff));
      background: -webkit-linear-gradient(#ffffff, #ffffff 85%, #ffffff);
      background: -moz-linear-gradient(#ffffff, #ffffff 85%, #ffffff);
      background: -o-linear-gradient(#ffffff, #ffffff 85%, #ffffff);
      background: -ms-linear-gradient(#ffffff, #ffffff 85%, #ffffff);
      background: linear-gradient(#ffffff, #ffffff 85%, #ffffff);
      background-color: white;
    }
    

7. I'm still looking for control over the font size. The obvious 

    .slides > article {
        font-size: 30px;
    }

doesn't do it. I'll keep searching, but if someone knows the answer feel free to chime in. 

8. I'm also looking for control over the slide title font, to make it bold/darker.

**Comment by ptoche** [comment]
As a side remark, I think Google has a newer version of io2012 than the one used in rCharts, though I do not know if there are any important differences between them.

**Comment by ptoche** [comment]
Another thing I'm noticing is that the slides load fast whenever I have a very good internet connection or whenever the internet is cut off entirely. But if I'm connected with a weak wifi signal, loading takes 10 or 20 or 30 seconds, displaying messages about google fonts or mathjax, reflecting the page's attempts to dowload the fonts/whatever from the internet, rather than using local resources. I wonder if it would is possible to completely by-pass the internet by saving fonts and such in local folders once and not every time?

**Comment by ramnathv** [comment]
@ptoche here are some quick thoughts. I will update with a more detailed reply next week.

1. `ios2012` should be able to grab assets from `assets/css`. If it is not working, it is a bug and I will look into it.
2. `io2013` is the newer version. There are minimal changes. It is easy to incorporate it.
3. You can save the slides along with a local copy of all dependencies. You need to set `mode: selfcontained` in your slide YAML and that will take care of it.

**Comment by ptoche** [comment]
Thanks a lot for the feedback Ramnath.

    1. I couldn't get ``io2012`` to grap ``css``  from ``assets/css``. However, on reflection that may not be a bad thing. Because I have a main index based on the ``minimal`` framework, some slides in ``io2012`` and (during a transition period of a few hours) ``html5slides`` all at once in the same project, the styles were interfering. For instance, my font size for the title intended for ``html5slides`` was read by ``minimal`` (which does source ``assets/css`` as expected) with a very bad result. The problem was fixed when I got rid of ``html5slides`` altogether in the project. I'm glad that my ``io2012`` css is currently isolated, so it doesn't get read by my ``minimal``-based general index.

    2. I had copied the ``io2013`` files as a brief experiment, but something went wrong when I overwrote the ``js`` folder, so I reverted back. Anyhow, I'm very happy with the current ``slidify`` implementation of ``io2012``  so I won't bother to try ``io2013`` unless it comes with ``slidify``.

    3. I have ``mode: selfcontained``  in the yaml already and the ``html`` output file's head is good:

        <head>
          <title>Introduction to Piketty's Capital</title>
          <meta charset="utf-8"/>
          <meta name="description" content="Introduction to Piketty's Capital"/>
          <meta name="author" content="Patrick Toche"/>
          <meta name="generator" content="slidify"/>
          <meta name="apple-mobile-web-app-capable" content="yes"/>
          <meta http-equiv="X-UA-Compatible" content="chrome=1"/>
          <link rel="stylesheet" href="../../libraries/frameworks/io2012/css/default.css" media="all" />
          <link rel="stylesheet" href="../../libraries/frameworks/io2012/css/phone.css" 
            media="only screen and (max-device-width: 480px)"/>
          <link rel="stylesheet" href="../../libraries/frameworks/io2012/css/slidify.css"/>
          <link rel="stylesheet" href="../../libraries/frameworks/io2012/css/custom.css"/>
          <link rel="stylesheet" href="../../libraries/highlighters/highlight.js/css/tomorrow.css" />
          <base target="_blank">   <link rel=stylesheet href="../../libraries/widgets/nvd3/css/nv.d3.css"></link>
        <link rel=stylesheet href="../../libraries/widgets/nvd3/css/rNVD3.css"></link>
        <link rel=stylesheet href="./assets/css/ribbons.css"></link>

 However, further down there are the following scripts, which I believe (based on the browser's messages while frozen) are causing the 'slow-ness' when the wifi signal is weak:

            <script 
              src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
            </script>

and:

          <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

For your information, the project I'm working on is here (of which I have only completed the Introduction and Chapter 01), if you want to peruse more code:

https://github.com/ptoche/piketty
http://ptoche.github.io/

I'll be happy to help in debugging if I can.

And a big thank you for your wonderful packages!

Patrick.

**Comment by ramnathv** [comment]
@ptoche Mathjax should be loading directly from your `libraries` folder. The script you see is a fallback that is trigerred only when a local copy is not found. The `CFInstall.min` script is puzzling since I don't think it is a part of the `io2012` framework. Do you see it in the html file you produce, or did you find it while doing `View Source` in your browser?

**Comment by ptoche** [comment]
@ramnathv, thanks for the feedback Ramnath. 

Yes, reference to the ``CFInstall.min`` file is from inside the ``html`` produced by slidifying the ``rmd`` source.  

If your interpretation is correct, then it would suggest that I do not have the local copy properly set up. However, the slides display properly: I cannot say if it's sourcing the intended Google fonts or some fallback alternative, but I can say that Mathjax is displaying correctly. However, I think the logic is the reverse: first the ``html`` looks for the remote files on the internet, only then does it fall back to the local copy if it cannot find it. Thus, if the internet is completely disconnected it fall backs quickly, but if the internet connection is sluggish it takes 20 or 30 seconds for it to realise that it's hopeless and fall back on the local copy after that delay (maybe there's a timer or it waits for an error message).  The reason I believe the second interpretation is correct is that the ``html``  contains the following comment:

    

You can view the ``html`` produced by slidify with the ``io2012`` framework here:

https://raw.githubusercontent.com/ptoche/piketty/master/lectures/Chapter%2000/index.html

The ``rmd``  source is here:

https://raw.githubusercontent.com/ptoche/piketty/master/lectures/Chapter%2000/index.Rmd

And for completeness, the ``html`` produced by ``bootstrap`` is here:

https://raw.githubusercontent.com/ptoche/piketty/master/index.html

and its ``rmd`` source here:

https://raw.githubusercontent.com/ptoche/piketty/master/index.Rmd

**Comment by ptoche** [comment]
I started off with the ``lectureSite`` template (see http://slidify.userecho.com/topic/577030-getting-started-with-lecturesite/), but later replaced the ``assets`` and ``frameworks`` and everything else from the github template of ``slidify`` by downloading the zip and copy-pasting the newer files into the appropriate directories. So unless I've messed up, I should have the latest setup.

**Comment by ptoche** [comment]
Is my interpretation correct?

**Comment by ramnathv** [comment]
I think you are right @ptoche. Let me dig deeper this weekend.

**Comment by ptoche** [comment]
Thanks Ramnath. There's absolutely no urgency, as I mention above I got the css to work and the solution if the wifi signal is weak is simply to cut it off completely. Cheers :-)

**Comment by rossellab** [comment]
I have the same issue with my custom css file in io2012, if I store it in assets/css it does not load. Also I can't get the workaround suggested by @ptoche to work. I'm using Slidify 0.5 (installed just today) and slidifyLibraries 0.3.1 from RStudio 0.98.1103 on a Mac OSX 10.6.8.
My css commands work if I include it in the first slide between html style tags.

**Comment by ptoche** [comment]
@rossellab, you can view the entire architecture of my project at https://github.com/ptoche/piketty, and then tediously check how your setup differs from mine? Good luck!

**Comment by rossellab** [comment]
Thank you! I'll try that way.
